### PR TITLE
Enhancements: Back to ToC links + other minor adjustments

### PR DESCRIPTION
**Pull request by dami-i**
- Adds "Back to Table of Contents" links on every section for navigation improvement;
- Moves a duplicate "step 4" on Visa payments to a side note;
- Standardizes the presentation for most of the steps sequences to bold text with colon as separator (e.g. "**Step 1:**");
- Formats Linux commands as monospace;
- Promotes a few texts to `h4` headers;
- Other minor adjustments.

Thank you very much for this free content.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,12 +149,12 @@ Architecture styles define how different components of an application programmin
 
   Notifies systems when events occur
 
+<sub>⬆️ [_Back to Table of Contents_](#table-of-contents)</sub>
```
Comment (slam) [inline README.md:152]: We like the idea, but we'd like to explore other ideas with our team on how to execute on this.

We'd like to accept the rest of the PR, so if you could remove the "Back to Table of Contents" links and resolve the current conflicts with main, that would be great.

Thank you for contributing.

Comment (max397574) [reply README.md:152]: Couldn't users just use the sticky TOC at the top left or right (depeding whether you use beta repo view or not)
image

Comment (dami-i) [reply README.md]: > We like the idea, but we'd like to explore other ideas with our team on how to execute on this.
> 
> We'd like to accept the rest of the PR, so if you could remove the "Back to Table of Contents" links and resolve the current conflicts with main, that would be great.
> 
> Thank you for contributing.

Done!
Thank you for your attention.

File: README.md [hunk 2]
```diff
@@ -874,14 +860,13 @@ Below you will find a diagram showing the microservice tech stack, both for the
   <img src="images/microservice-tech.jpeg" />
 </p>
 
-
-▶️ 𝐏𝐫𝐞-𝐏𝐫𝐨𝐝𝐮𝐜𝐭𝐢𝐨𝐧
+#### Pre-production
```
Comment (slam) [inline README.md:863]: I don't think we want to change these into sub-headings. It makes the outline view super busy:

image

File: README.md [hunk 3]
```diff
@@ -1042,7 +1022,7 @@ A k8s cluster consists of a set of worker machines, called nodes, that run conta
 
 The worker node(s) host the Pods that are the components of the application workload. The control plane manages the worker nodes and the Pods in the cluster. In production environments, the control plane usually runs across multiple computers, and a cluster usually runs multiple nodes, providing fault tolerance and high availability.
 
-- Control Plane Components
+#### Control Plane Components
```
Comment (slam) [inline README.md:1025]: Another example of busy outline view:

image

**Review by slam — CHANGES_REQUESTED** [review]

**Comment by dami-i** [comment]
Closing due to unavailability of time to tackle the changes.

**Pull request closed without merging**